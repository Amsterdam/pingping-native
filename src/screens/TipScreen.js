import React from 'react';
import {StyleSheet, View} from 'react-native';
import SimpleHeader from '../components/header/SimpleHeader';
import ContentLayout from '../components/layout/ContentLayout';
import {Content, Container} from 'native-base';
import Title from '../components/typography/Title';
import Body from '../components/typography/Body';
import OvalDefault from '../components/layout/OvalDefault';

const styles = StyleSheet.create({
  title: {
    marginBottom: 25,
  },
  subTitle: {
    fontSize: 18,
    marginBottom: 10,
  },
  paragraphContainer: {
    marginBottom: 25,
  },
});

const TipScreen = ({navigation}) => {
  return (
    <Container>
      <OvalDefault />
      <SimpleHeader navigation={navigation} color="white" />
      <ContentLayout>
        <Content>
          <Title style={styles.title}>Tips</Title>
          <View style={styles.paragraphContainer}>
            <Title style={styles.subTitle}>Hoeveel abonnementen heb jij?</Title>
            <Body>
              Automatische afschrijvingen kunnen handig zijn, maar ook
              gevaarlijk! Als je niet gebruik maakt van abonnementen zoals de
              sportschool, swapfiets of zoiets, stop dan je abonnement. Zo loopt
              het niet door en bespaar je geld.
            </Body>
          </View>
          <View style={styles.paragraphContainer}>
            <Title style={styles.subTitle}>
              Zet je eigen goal om te sparen!
            </Title>
            <Body>
              Een handige manier om te sparen; de 52 weken challenge! Spaar in
              de eerste week 1euro en spaar vervolgens elke week een euro meer.
              Hou je dit 52 weken vol dan heb je op een makkelijke manier 1378
              euro gespaard in een jaar!
            </Body>
          </View>
          <View style={styles.paragraphContainer}>
            <Title style={styles.subTitle}>Spaar je eigen risico</Title>
            <Body>
              Ga bij je Zorgverzekeraar na of je je eigen risico verspreid over
              het jaar kunt betalen. Moet je gebruik maken van je verzekering
              omdat je ziektekosten hebt, dan sta je niet voor verassingen. Maak
              je geen gebruik van ziektekosten dan kun je je betaalde eigen
              risico over dat jaar terugkrijgen! Een fijn spaarpotje!
            </Body>
          </View>
          <Body>Kom je er niet uit? Vraag om hulp in de!</Body>
        </Content>
      </ContentLayout>
    </Container>
  );
};

export default TipScreen;
