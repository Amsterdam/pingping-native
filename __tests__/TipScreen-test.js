import React from 'react';

import renderer from 'react-test-renderer';

import TipScreen from '../src/screens/TipScreen';

jest.useFakeTimers();
const navigation = {navigate: jest.fn()};
const route = {params: {tips: []}};

test('renders correctly', () => {
	const tree = renderer
		.create(<TipScreen navigation={navigation} route={route} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
