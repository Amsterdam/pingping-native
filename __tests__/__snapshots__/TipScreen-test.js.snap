// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        noShadow={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "transparent",
              "borderBottomWidth": 0.5,
              "elevation": 0,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 6,
              "paddingRight": 10,
              "paddingTop": undefined,
              "right": 0,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
              "top": 0,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            false,
          ]
        }
        transparent={true}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            forwardedRef={[Function]}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginLeft": -3,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 4,
                "paddingRight": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID="HEADER_BACK_BUTTON"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 34,
                      "marginLeft": 1,
                      "marginRight": 2,
                      "marginTop": 0,
                      "paddingTop": 1,
                    },
                    Object {
                      "color": "#000",
                    },
                  ],
                  Object {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="AntDesign"
            >
              î˜œ
            </Text>
          </View>
        </View>
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "Heavitas",
              },
              Object {
                "fontSize": 26,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "#fb9f00",
                "marginTop": 5,
              },
            ]
          }
          uppercase={false}
        >
          Tips
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "alignSelf": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              },
              Object {
                "flex": 1,
              },
            ]
          }
        />
      </View>
    </RCTSafeAreaView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 25,
        "position": "relative",
        "width": 750,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Array [
            Object {
              "padding": undefined,
            },
            Object {
              "alignItems": "center",
              "margin": 10,
            },
          ]
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableOnAndroid={false}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollIntoView={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          }
        }
        update={[Function]}
        viewIsInsideTabBar={false}
      >
        <View />
      </RCTScrollView>
    </RCTSafeAreaView>
  </View>
</View>
`;
